pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -266,6 +274,30 @@ func ingressBackendToHTTPRoute(backend *v1beta1.IngressBackend, namespace string
 	}
 }
 
+func resolveNamedPort(backend *v1beta1.IngressBackend, namespace string, client kubernetes.Interface) int32 {
+	svc, err := client.CoreV1().Services(namespace).Get(context.TODO(), backend.ServiceName, metaV1.GetOptions{})
```

Comment:
We already have an informer, we should probably be using this. Otherwise all of these calls and list are going to be synchronous and kill the api server. We may need to do some refactoring mentioned in https://github.com/istio/istio/issues/24640 though

Reply:
You are right, we could reuse svc informer used by serviceregistry controller and pods from another controller. Could this wait and maybe I could start working on refactoring informers on a separate PR?

Reply:
Bump, @howardjohn what do you think?

Reply:
I would just create one from https://github.com/istio/istio/blob/b576237ad1280f5abc17291d84b1a913032b7183/pilot/pkg/config/kube/ingress/controller.go#L147 for now then in another PR we can share the same maybe?

Reply:
Makes sense, will add informers for service and pods in ingress controller. 

Also convert-ingress is called from istio-cmd, I'm guessing it would pass a similar interface as informer store?
https://github.com/istio/istio/blob/1a25f900edaafa96e9cda6571c26070832004bbb/istioctl/cmd/convert_ingress.go#L68

Reply:
This is done :-)

Reply:
@howardjohn I have made the requested changes, please do review when you get a chance